simln-lib/src/test_utils.rs

```diff
@@ -45,3 +54,187 @@ pub fn create_nodes(n: usize, node_capacity: u64) -> Vec<(NodeInfo, u64)> {
+);
+
+/// A builder for creating mock Lightning nodes for testing purposes.
+///
+/// This struct provides a flexible way to configure test nodes with features like
+/// keysend support and specific networks. It supports both full node setups (with
+/// node info and clients) and client-only setups for network testing.
+///
+/// # Examples
+/// Basic setup with keysend on specific nodes:
+/// let (nodes, clients) = TestNetworkBuilder::setup_test_nodes(5, &[0, 2]);
+///
+/// Using the builder for more control:
+/// let (nodes, clients) = TestNetworkBuilder::new(5)
+///     .with_keysend_nodes(vec![0, 2, 4])
+///     .build_full();
+///
+/// Building clients with specific networks:
+/// let clients = TestNetworkBuilder::new(3)
+///     .with_networks(vec![Network::Bitcoin, Network::Testnet, Network::Regtest])
+///     .build_clients_only();
+pub struct LightningTestNodeBuilder {
+    node_count: usize,              // Required - must be provided at creation
+    initial_balance: u64,           // Always has a value (default: 100,000)
+    keysend_indices: Vec<usize>,    // Always a vector (default: empty)
+    networks: Option<Vec<Network>>, // Can be None (default) or Some(networks)
+}
+
+impl LightningTestNodeBuilder {
+    /// Creates test nodes with optional keysend support on specified nodes.
+    /// A convenience method equivalent to `new(node_count).with_keysend_nodes(indices).build_full()`.
+    pub fn setup_test_nodes(node_count: usize, keysend_indices: &[usize]) -> TestNodesResult {
+        Self::new(node_count)
+            .with_keysend_nodes(keysend_indices.to_vec())
+            .build_full()
+    }
+
+    /// Creates test nodes with specified networks.
+    /// A convenience method equivalent to `new(node_count).with_networks(networks).build_clients_only()`.
+    pub fn setup_network_test_nodes(
+        node_count: usize,
+        networks: Vec<Network>,
+    ) -> HashMap<PublicKey, Arc<Mutex<dyn LightningNode>>> {
+        Self::new(node_count)
+            .with_networks(networks)
+            .build_clients_only()
+    }
+
+    /// Creates a new builder instance with the specified number of nodes.
+    /// The default configuration includes a balance of 100,000 units per node,
+    /// no keysend support, and no specific networks.
+    pub fn new(node_count: usize) -> Self {
+        Self {
+            node_count,
+            initial_balance: 100_000,    // Default 100k sats
+            keysend_indices: Vec::new(), // No keysend by default
+            networks: None,              // No specific networks by default
+        }
+    }
+
+    /// Specifies which nodes should have the keysend feature enabled
+    pub fn with_keysend_nodes(mut self, indices: Vec<usize>) -> Self {
+        self.keysend_indices = indices;
+        self
+    }
+
+    /// Sets specific networks for each node
+    /// Will panic if the number of networks doesn't match node_count
+    /// Returns self for method chaining
```

Comment:
nit: here and a few other places - wrap at 120 and end comments with full stops.